src/zui/ZUIHeader/index.tsx

```diff
@@ -0,0 +1,280 @@
+import { OverridableComponent } from '@mui/material/OverridableComponent';
+import {
+  Avatar,
+  Box,
+  Divider,
+  ListItemIcon,
+  ListItemText,
+  MenuItem,
+  MenuList,
+  Popover,
+  SvgIconTypeMap,
+  Typography,
+  useTheme,
+} from '@mui/material';
+import { ExpandMore, MoreVert } from '@mui/icons-material';
+import { FC, useState } from 'react';
+
+import ZUIButton from 'zui/ZUIButton';
+import ZUIButtonGroup from 'zui/ZUIButtonGroup';
+import ZUIMenu, { MenuItem as MenuItemType } from 'zui/ZUIMenu';
+import ZUIEditTextinPlace from 'zui/ZUIEditTextInPlace';
+import ZUIBreadcrumbs, { BreadcrumbTreeItem } from 'zui/ZUIBreadcrumbs';
+import { WithRequired } from 'utils/types';
+
+interface ZUIHeaderProps {
+  /**
+   * Either an array of menu items, to be displayed as a menu OR a function that returns a component that will be displayed in the popover.
+   * If the action button should open a menu - send in menu items, not a whole component.
+   */
+  actionButtonPopoverContent?:
+    | WithRequired<MenuItemType, 'startIcon'>[]
+    | ((onClose: () => void) => JSX.Element);
+
+  /**
+   * The text on the action button
+   */
+  actionButtonLabel?: string;
+
+  /**
+   * Type of actionbutton. Defaults to 'secondary'
+   */
+  actionButtonType?: 'primary' | 'secondary';
```

Comment:
This might change if we decide to change the name of the relevant property on `ZUIButton`.